### PR TITLE
Fix crash in ~GLSLWaveformWidget()

### PR DESCRIPTION
**Pull request by uklotzde**
https://bugs.launchpad.net/mixxx/+bug/1877487
https://bugs.launchpad.net/mixxx/+bug/1856888

This at least fixes the Alt+F4 crash on Linux.

### CODE REVIEW

File: src/waveform/widgets/glslwaveformwidget.cpp [hunk 1]
```diff
@@ -64,12 +64,8 @@ GLSLWaveformWidget::GLSLWaveformWidget(
     m_initSuccess = init();
 }
 
-GLSLWaveformWidget::~GLSLWaveformWidget() {
-    makeCurrent();
-}
-
 void GLSLWaveformWidget::castToQWidget() {
-    m_widget = static_cast<QWidget*>(static_cast<QGLWidget*>(this));
+    m_widget = this;
```
Comment (Be-ing) [inline src/waveform/widgets/glslwaveformwidget.cpp:68]: Is this change necessary? ba340abe491f5dd7e9e8b4cd5296ab64e21d7a98#diff-022b8a0f41d44be564c4ef50a719a09cL56 did not do it for the other GL waveforms.

Comment (Be-ing) [reply src/waveform/widgets/glslwaveformwidget.cpp:68]: It builds, so I guess it's okay.

**Comment by Be-ing** [comment]
Thank you!

File: src/waveform/widgets/glslwaveformwidget.cpp (hunk 1 above)
Comment (uklotzde) [reply src/waveform/widgets/glslwaveformwidget.cpp:68]: Needless redundancy. IDK why this was necessary.

**Comment by ronso0** [comment]
Unfortunately it still crashes when reloading Deere.
